unblob/models.py

```diff
@@ -81,21 +85,40 @@ def __repr__(self):
         return self.range_hex
 
 
-class Handler(Protocol):
+class Handler(abc.ABC):
     """A file type handler is responsible for searching, validating and "unblobbing" files from Blobs."""
 
     NAME: str
     YARA_RULE: str
     # We need this, because not every match reflects the actual start
     # (e.g. tar magic is in the middle of the header)
-    YARA_MATCH_OFFSET: int
+    YARA_MATCH_OFFSET: int = 0
 
-    @staticmethod
+    C_STRUCTURES: Optional[str] = None
+
+    # One of the C_STRUCTURES used to parse the file's header
+    HEADER_STRUCT: Optional[str] = None
+
+    def __init__(self):
+        if self.C_STRUCTURES:
+            self.cparser = cstruct()
+            self.cparser.load(self.C_STRUCTURES)
+
+            if self.HEADER_STRUCT:
+                self._header_parser = getattr(self.cparser, self.HEADER_STRUCT)
+
+    def parse_header(self, file: io.BufferedIOBase):
```

Comment:
If we add an optional endianness argument that defaults to little endian, we could rely on parse_header in CPIO v3 handlers, and be ready for future use cases where we need to handle big endian transparently without rewriting parse_header in the subclass.

```suggestion
    def parse_header(self, file: io.BufferedIOBase, endian= "<"):
      self.cparser.endian = endian
```

Reply:
There will be only one `Handler` class instance during the whole run, so we still must not change any state of the instance. The __init__ is there, so the subclasses can customize some common behavior or initialize extra attributes needed.

Reply:
What is the overhead of `cparser` initialization? Is it worth storing it in the instance?

Reply:
```
>>> In [10]: %timeit parser=cstruct(); parser.load(C_STRUCTURES)
2.17 ms ± 63.1 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Reply:
Ok that makes sense. Let's leave it this way then.